hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_utils.py

```
@@ -76,22 +46,43 @@ class Config:
         title = "bloxroute_openbook"
 
 
-KEYS = BloxRouteConnectorMap.construct()
-
-
-# def TradeTypeToSide(type: TradeType) -> api.Side:
-#     if type.value == type.BUY:
-#         return api.Side.S_BID
-#     elif type.value == type.SELL:
-#         return api.Side.S_ASK
-#     else:
-#         return api.Side.S_UNKNOWN
-#
-#
-# def OrderTypeToBlxrOrderType(orderType: OrderType) -> api.OrderType:
-#     if orderType.value == orderType.MARKET:
-#         return api.OrderType.OT_MARKET
-#     elif orderType.value == orderType.LIMIT:
-#         return api.OrderType.OT_LIMIT
-#     else:
-#         raise Exception(f"unknown order type ${orderType.value}")  # TODO need unknown value
+KEYS = BloXrouteConnectorMap.construct()
+
+
+def trade_type_to_side(trade_type: TradeType) -> api.Side:
+    if trade_type == TradeType.BUY:
+        return api.Side.S_BID
+    elif trade_type == TradeType.SELL:
+        return api.Side.S_ASK
+    else:
+        return api.Side.S_UNKNOWN
+
+
+def order_type_to_blxr_order_type(order_type: OrderType) -> common.OrderType:
+    if order_type == OrderType.MARKET:
+        return common.OrderType.OT_MARKET
+    elif order_type == OrderType.LIMIT:
+        return common.OrderType.OT_LIMIT
+    else:
+        raise Exception(f"unknown order type {order_type.value}")
+
+# gets the last n digits of a number
+def truncate(num: int, n: int) -> int:
+    num_str = str(num)
+    trunc_num_str = num_str[-n:]
+    return int(trunc_num_str)
+
+T = typing.TypeVar("T")
+
+
+async def retry(f: Callable[..., Coroutine[Any, Any, T]], check_field: str, retries: int) -> T:
+    for i in range(retries):
+        response = await f()
+        if not isinstance(response, betterproto.Message):
+            raise Exception("response does not implement betterproto.Message")
+
+        response_dict = response.to_dict()
+        if check_field in response_dict and response_dict[check_field]:
```

when are you seeing this happen such that this is necessary? In the Python SDK we should throw an exception if we get back an error. It's a bit of a bad look to have methods in an SDK that you have to do this kind of thing, since that indicates that the SDK itself isn't reliable in doing what its functions expect to.